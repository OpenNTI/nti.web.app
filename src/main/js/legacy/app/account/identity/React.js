const Ext = require('@nti/extjs');
const { getService } = require('@nti/web-client');
const { Identity } = require('@nti/web-profiles');
const LoginActions = require('internal/legacy/login/Actions');
require('internal/legacy/overrides/ReactHarness');

const AccountActions = require('../Actions');

module.exports = exports = Ext.define(
	'NextThought.app.account.identity.React',
	{
		extend: 'NextThought.ReactHarness',
		alias: 'widget.identity-react',
		component: Identity.Session,

		setTheme(scope, knockout) {},

		initComponent: function () {
			this.callParent(arguments);
			Object.assign(this.initialConfig, {
				addHistory: true,
				baseroute: '/app',
			});
			this.AccountActions = AccountActions.create();
			this.LoginActions = LoginActions.create();
		},

		async afterRender() {
			this.callParent(arguments);
			const service = await getService();
			const supportLinks = service.getSupportLinks();

			this.mon(Ext.getBody(), {
				click: this.onBodyClicks.bind(this, supportLinks),
			});
		},

		onBodyClicks(supportLinks, e) {
			let action = e.getTarget('a[data-action]')?.dataset.action;
			const link = e.getTarget('a[href]');
			const href = link?.getAttribute('href');

			// This is used to launch support logic the same way everywhere
			// we have a link that goes to the supportContact url. (so we can
			// just emit that url in the dom and not worry about coping logic)
			if (
				supportLinks.supportContact === href &&
				supportLinks.internalSupport
			) {
				action = action || 'support';
			}

			if (action) {
				if (call(this, `do-${action}`, link, e) !== false) {
					e.stopEvent();
				}
			}
		},

		showLink(target) {
			target.setAttribute('target', '_blank');
			return false;
		},

		'do-welcomeGuide'(target) {
			this.AccountActions.showWelcomePage(target.href);
		},

		'do-about': 'showLink',

		'do-privacy'(target) {
			this.AccountActions.showPrivacy(target);
		},

		'do-privacyForMinors'(target) {
			this.AccountActions.showChildrensPrivacy(target.href);
		},

		'do-terms'(target) {
			this.AccountActions.showTermsOfService(target.href);
		},

		'do-support'(target) {
			this.AccountActions.showContactUs();
		},

		'do-helpSite': 'showLink',

		'do-impersonate'() {
			this.LoginActions.handleImpersonate();
		},

		'do-logout'() {
			this.LoginActions.handleLogout();
		},
	}
);

function call(scope, action, ...args) {
	if (typeof action === 'string') {
		return call(scope, scope[action], ...args);
	}

	return action.apply(scope, args);
}
