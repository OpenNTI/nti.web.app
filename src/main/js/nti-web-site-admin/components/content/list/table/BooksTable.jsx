import React from 'react';
import PropTypes from 'prop-types';
import {Table, Loading} from '@nti/web-commons';
import {scoped} from '@nti/lib-locale';
import {searchable, contextual} from '@nti/web-search';

import {BookName, CreatedTime} from './columns';
import Store from './BooksStore';
import Pager from './Pager';
import EmptyState from './EmptyState';

const t = scoped('nti-web-site-admin.content.list.table.BooksTable', {
	title: 'Books',
	emptyMessage: 'There are no books'
});

export default
@searchable()
@contextual(t('title'))
@Store.connect({
	loading: 'loading',
	items: 'items',
	error: 'error',
	sortOn: 'sortOn',
	sortDirection: 'sortDirection',
	pageNumber: 'pageNumber',
	numPages: 'numPages'
})
class BooksTable extends React.Component {
	static propTypes = {
		store: PropTypes.object.isRequired,
		items: PropTypes.array,
		error: PropTypes.string,
		loading: PropTypes.bool,
		sortOn: PropTypes.string,
		sortDirection: PropTypes.string,
		pageNumber: PropTypes.number,
		numPages: PropTypes.number
	}

	items = []

	columns = [
		BookName,
		CreatedTime
	]

	state = {
		sortDirection: Table.ASCENDING,
		sortOn: 'name'
	}

	onSortChange = (sortKey, sortDirection) => {
		this.props.store.setSort(sortKey, sortDirection);
	}

	renderControls () {
		return (
			<div className="controls">
				{/* <div className="button change-role" onClick={this.showChangeRolesDialog}>{t('changeRole')}</div> */}
			</div>
		);
	}

	renderHeader () {
		return (
			<div className="header">
				<div className="title">{t('title')}</div>
			</div>
		);
	}

	render () {
		const {store, sortOn, sortDirection, items, error, loading, numPages, pageNumber} = this.props;

		return (
			<div className="content-table-container">
				{loading && <Loading.Mask/>}
				{!loading && error && <div className="error">{error}</div>}
				{!error && !loading && (!items || items.length === 0) && <EmptyState message={t('emptyMessage')}/>}
				{!loading && items && items.length > 0 && (
					<div>
						{this.renderHeader()}
						<Table.Table
							store={store}
							items={items || []}
							sortDirection={sortDirection}
							sortOn={sortOn}
							columns={this.columns}
							onSortChange={this.onSortChange}
						/>
						<Pager store={store} numPages={numPages} pageNumber={pageNumber}/>
					</div>
				)}
			</div>
		);
	}
}
