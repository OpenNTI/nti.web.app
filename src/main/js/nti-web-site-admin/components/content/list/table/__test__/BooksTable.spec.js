import React from 'react';
import renderer from 'react-test-renderer';
import { TestUtils } from '@nti/web-client';

import BooksTable from '../BooksTable';

const { tearDownTestClient, setupTestClient } = TestUtils;

const getMockService = numberOfBooks => {
	return {
		getCollection: () => {
			return {
				hasLink: () => true,
				getLink: () => 'mockLink',
			};
		},
		getBatch: () => {
			let titles = [];

			for (let i = 0; i < numberOfBooks; i++) {
				const index = i + 1;

				titles.push('id' + index);
			}

			return Promise.resolve({
				Total: numberOfBooks,
				BatchPage: 1,
				titles,
			});
		},
		getWorkspace: () => {
			return {
				getLink: () => 'mockLink',
			};
		},
		getObject: id => {
			return {
				title: id,
				getID: () => id,
				getCreatedTime: () => new Date('10/30/2017'),
				getDefaultAssetRoot: () => 'testRoot',
			};
		},
	};
};

const onBefore = numberOfBooks => {
	jest.useFakeTimers();
	setupTestClient(getMockService(numberOfBooks));
};

const onAfter = () => {
	tearDownTestClient();
};

const flushPromises = () => new Promise(resolve => setImmediate(resolve));

/* eslint-env jest */
describe('Site admin content book list (with no items)', () => {
	beforeEach(() => onBefore(0));
	afterEach(onAfter);

	test('Basic render test', async () => {
		const cmp = renderer.create(<BooksTable />);

		jest.runAllTimers();
		await flushPromises();
		jest.runAllTimers();

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});

describe('Site admin content book list (with 5 items)', () => {
	beforeEach(() => onBefore(5));
	afterEach(onAfter);

	test('Basic render test', async () => {
		const cmp = renderer.create(<BooksTable />);

		jest.runAllTimers();
		await flushPromises();
		jest.runAllTimers();

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});

describe('Site admin content book list (with 25 items)', () => {
	beforeEach(() => onBefore(25));
	afterEach(onAfter);

	test('Basic render test', async () => {
		const cmp = renderer.create(<BooksTable />);

		jest.runAllTimers();
		await flushPromises();
		jest.runAllTimers();

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
