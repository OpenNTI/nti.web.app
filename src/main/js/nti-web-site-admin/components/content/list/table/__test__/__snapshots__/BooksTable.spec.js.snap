// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site admin content book list (with 5 items) Basic render test 1`] = `
<div
  className="content-table-container"
>
  <div>
    <div
      className="header"
    >
      <div
        className="title"
      >
        Books
      </div>
    </div>
    <table
      className="nti-generic-table"
    >
      <thead>
        <tr>
          <th
            className="bookname-col nti-table-simple-header"
          >
            <div
              className="sortable"
              onClick={[Function]}
            >
              <span>
                Name
              </span>
            </div>
          </th>
          <th
            className="createdtime-col nti-table-simple-header"
          >
            <div
              className="sortable"
              onClick={[Function]}
            >
              <span>
                Created
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id1"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id1
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id2"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id2
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id3"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id3
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id4"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id4
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id5"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id5
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Site admin content book list (with 25 items) Basic render test 1`] = `
<div
  className="content-table-container"
>
  <div>
    <div
      className="header"
    >
      <div
        className="title"
      >
        Books
      </div>
    </div>
    <table
      className="nti-generic-table"
    >
      <thead>
        <tr>
          <th
            className="bookname-col nti-table-simple-header"
          >
            <div
              className="sortable"
              onClick={[Function]}
            >
              <span>
                Name
              </span>
            </div>
          </th>
          <th
            className="createdtime-col nti-table-simple-header"
          >
            <div
              className="sortable"
              onClick={[Function]}
            >
              <span>
                Created
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id1"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id1
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id2"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id2
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id3"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id3
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id4"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id4
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id5"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id5
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id6"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id6
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id7"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id7
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id8"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id8
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id9"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id9
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id10"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id10
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id11"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id11
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id12"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id12
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id13"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id13
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id14"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id14
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id15"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id15
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id16"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id16
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id17"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id17
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id18"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id18
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id19"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id19
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id20"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id20
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id21"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id21
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id22"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id22
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id23"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id23
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id24"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id24
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="bookname-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="./book/id25"
            >
              <div
                className="cell"
              >
                <img
                  className="book-icon"
                  src="contentpackage-landing-232x170.png?t=0"
                />
                <div
                  className="book-info"
                >
                  <div
                    className="title"
                  >
                    id25
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="createdtime-col"
          >
            <div
              className="cell"
            >
              <time
                className="value ntiText"
                dateTime="2017-10-30T00:00:00Z"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
    <div
      className="site-admin-table-pager"
    >
      <div
        className="page selected"
        onClick={[Function]}
      >
        1
      </div>
      <div
        className="page"
        onClick={[Function]}
      >
        2
      </div>
    </div>
  </div>
</div>
`;

exports[`Site admin content book list (with no items) Basic render test 1`] = `
<div
  className="content-table-container"
>
  <div
    className="empty-state"
  >
    There are no books
  </div>
</div>
`;
